src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java

```diff
@@ -34,17 +47,37 @@ public void configure(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
             .name("Record listing")
             .showSummary()
             .description("Record listing")
+            .aboutToStartCallback(this::aboutToStart)
             .aboutToSubmitCallback(this::aboutToSubmit)
             .submittedCallback(this::submitted)
             .showEventNotes()
             .grant(CREATE_READ_UPDATE_DELETE, COURT_ADMIN_CIC, SUPER_USER)
             .grantHistoryOnly(SOLICITOR));
 
         addHearingTypeAndFormat(pageBuilder);
+        addRegionInfo(pageBuilder);
+        hearingVenues.addTo(pageBuilder);
         addRemoteHearingInfo(pageBuilder);
         addOtherInformation(pageBuilder);
     }
 
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(CaseDetails<CaseData, State> details) {
+        var caseData = details.getData();
+        DynamicList regionList = locationService.getAllRegions();
+        caseData.getRecordListing().setRegionList(regionList);
+
+        if (regionList.getListItems().isEmpty()) {
+            caseData.getRecordListing().setRegionsMessage("Unable to retrieve Region data");
```

Comment:
Style thing, but could use ternary operator here?

`if (regionList.getListItems().isEmpty()) ? caseData.getRecordListing().setRegionsMessage("Unable to retrieve Region data") : caseData.getRecordListing().setRegionsMessage("");`

Reply:
Done. Used ternary operator now,